### PR TITLE
Move  Edit.js Tool Window Tabs to the Top

### PR DESCRIPTION
**Pull request by imgntn**
This PR moves the tabs in the edit.js tool window to the top of the screen and removes the redundant title bar -- it was easy to lose them before

capture

### CODE REVIEW

File: interface/src/scripting/WebWindowClass.cpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ WebWindowClass::WebWindowClass(const QString& title, const QString& url, int wid
 
         dockWidget->setWidget(_webView);
 
-        toolWindow->addDockWidget(Qt::RightDockWidgetArea, dockWidget);
+		toolWindow->addDockWidget(Qt::TopDockWidgetArea, dockWidget, Qt::Horizontal);
```
Comment (ZappoMan) [inline interface/src/scripting/WebWindowClass.cpp]: indenting is wrong... looks like you inserted TABS here... these should be spaces...

File: interface/src/ui/ToolWindow.cpp [hunk 2]
```diff
@@ -118,7 +118,18 @@ void ToolWindow::addDockWidget(Qt::DockWidgetArea area, QDockWidget* dockWidget)
 }
 
 void ToolWindow::addDockWidget(Qt::DockWidgetArea area, QDockWidget* dockWidget, Qt::Orientation orientation) {
-    QMainWindow::addDockWidget(area, dockWidget, orientation);
+	QList<QDockWidget*> dockWidgets = findChildren<QDockWidget*>();
```
Comment (ZappoMan) [inline interface/src/ui/ToolWindow.cpp]: indenting is wrong

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7a8db803eb2199bed4a30c97ec7e3df355d6bf0e available at: https://gist.github.com/57af2b1efb34f6487ee7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.dmg

File: interface/src/scripting/WebWindowClass.cpp (hunk 1 above)
Comment (huffman) [inline interface/src/scripting/WebWindowClass.cpp]: I'm not sure this line needed to be changed, although I could be wrong. The QMainWindow has different docking areas, each which can have its own set of tabs. It shouldn't which docking area the widgets are in as long as they are all the same.

File: interface/src/ui/ToolWindow.cpp [hunk 3]
```diff
@@ -118,7 +118,18 @@ void ToolWindow::addDockWidget(Qt::DockWidgetArea area, QDockWidget* dockWidget)
...
+	
+	setTabPosition(Qt::TopDockWidgetArea, QTabWidget::TabPosition::North);
```
Comment (huffman) [inline interface/src/ui/ToolWindow.cpp]: Do you need to set this every time a dock widget is added? I would think it would only need to be set at initialization (in the ctor).

File: interface/src/ui/ToolWindow.cpp [hunk 4]
```diff
@@ -118,7 +118,18 @@ void ToolWindow::addDockWidget(Qt::DockWidgetArea area, QDockWidget* dockWidget)
...
+
+	QMainWindow::addDockWidget(area, dockWidget, orientation);
+
+	// We want to force tabbing, so retabify all of our widgets.
+	QDockWidget* lastDockWidget = dockWidget;
+	foreach(QDockWidget* nextDockWidget, dockWidgets) {
+		tabifyDockWidget(lastDockWidget, nextDockWidget);
+		lastDockWidget = nextDockWidget;
+	}
```
Comment (huffman) [inline interface/src/ui/ToolWindow.cpp]: This is weird - why is this showing up in the diff? These lines were reindented, but the diff doesn't show them also being removed.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 815b263954da579feb22f49fd53ff2767cdc6a9c available at: https://gist.github.com/57af2b1efb34f6487ee7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.dmg

File: interface/src/ui/ToolWindow.cpp [hunk 5]
```diff
@@ -118,8 +119,20 @@ void ToolWindow::addDockWidget(Qt::DockWidgetArea area, QDockWidget* dockWidget)
 }
 
 void ToolWindow::addDockWidget(Qt::DockWidgetArea area, QDockWidget* dockWidget, Qt::Orientation orientation) {
+    QList<QDockWidget*> dockWidgets = findChildren<QDockWidget*>();
+    
+    setTabPosition(Qt::TopDockWidgetArea, QTabWidget::TabPosition::North);
```
Comment (huffman) [inline interface/src/ui/ToolWindow.cpp]: Have you tried putting this in the ctor only?

Comment (huffman) [inline interface/src/ui/ToolWindow.cpp]: If it doesn't work there, this probably needs to be added to the other addDockWidget method

**Comment by hifi-gustavo** [comment]
Build console output for commit: d45f48c8cd1663b9633278fb33c89e1af5523126 available at: https://gist.github.com/57af2b1efb34f6487ee7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 536351e9d5053d731bab08b3a910fb8d071379c6 available at: https://gist.github.com/57af2b1efb34f6487ee7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a1e2292e62e8daf155ff2711ca457f1e80e6dcb7 available at: https://gist.github.com/57af2b1efb34f6487ee7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c18aeef6b39a6b2c83a2813056876c851bfd96a6 available at: https://gist.github.com/57af2b1efb34f6487ee7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.dmg

**Comment by imgntn** [comment]
bump for review -- not sure why 'no test results found' ??

**Comment by hifi-gustavo** [comment]
Build console output for commit: 84973c590d2e5f9f7eb221c40f065782e91b0d29 available at: https://gist.github.com/57af2b1efb34f6487ee7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c8b6fed0e041de9e59192543a423e990f8614d19 available at: https://gist.github.com/57af2b1efb34f6487ee7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.dmg

**Comment by imgntn** [comment]
@huffman review?? :)

File: interface/src/ui/ToolWindow.cpp [hunk 6]
```diff
@@ -118,7 +118,18 @@ void ToolWindow::addDockWidget(Qt::DockWidgetArea area, QDockWidget* dockWidget)
 }
 
 void ToolWindow::addDockWidget(Qt::DockWidgetArea area, QDockWidget* dockWidget, Qt::Orientation orientation) {
+    QList<QDockWidget*> dockWidgets = findChildren<QDockWidget*>();
+
     QMainWindow::addDockWidget(area, dockWidget, orientation);
+    
+    QDockWidget* lastDockWidget = dockWidget;
+
+    lastDockWidget->setTitleBarWidget(new QWidget());
```
Comment (huffman) [inline interface/src/ui/ToolWindow.cpp]: This looks like a memory leak. You should pass `lastDockWidget` the QWidget ctor so it sets it as the parent. When the dock widget is cleaned up it will delete the QWidget. This is specific to Qt.

Comment (huffman) [inline interface/src/ui/ToolWindow.cpp]: Actually, you should put this in WebWindow class - we shouldn't be modifying the dock widget here.

File: interface/src/ui/ToolWindow.cpp [hunk 7]
```diff
@@ -37,11 +39,8 @@ bool ToolWindow::event(QEvent* event) {
             QMainWindow* mainWindow = qApp->getWindow();
             QRect mainGeometry = mainWindow->geometry();
 
-            int titleBarHeight = UIUtil::getWindowTitleBarHeight(this);
-            int topMargin = titleBarHeight;
-
-            _lastGeometry = QRect(mainGeometry.topLeft().x(),  mainGeometry.topLeft().y() + topMargin,
-                                  DEFAULT_WIDTH, mainGeometry.height() - topMargin);
+            _lastGeometry = QRect(mainGeometry.topLeft().x(),  mainGeometry.topLeft().y(),
+                                  DEFAULT_WIDTH, mainGeometry.height());
```
Comment (huffman) [inline interface/src/ui/ToolWindow.cpp]: Why was this removed? I'm not sure the DockWidget title bar changed would have affected this?

Comment (huffman) [inline interface/src/ui/ToolWindow.cpp]: Update: I just tested it - it looks like the tool window now extends over the main application title bar. The intended behavior was to have it start below it. I think we should revert this change unless there was a reason for it.

**Comment by hifi-gustavo** [comment]
Build console output for commit: aee11d4c06f1a7d26677d2a2f251090db499a3aa available at: https://gist.github.com/57af2b1efb34f6487ee7<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6078.dmg
